src/csharp/Grpc.Core.Api/IAsyncStreamReader.cs

```diff
@@ -50,7 +47,20 @@ namespace Grpc.Core
     /// </para>
     /// </summary>
     /// <typeparam name="T">The message type.</typeparam>
-    public interface IAsyncStreamReader<T> : IAsyncEnumerator<T>
+    public interface IAsyncStreamReader<T>
     {
+        /// <summary>
+        /// Gets the current element in the iteration.
+        /// </summary>
+        T Current { get; }
+
+        /// <summary>
+        /// Advances the reader to the next element in the sequence, returning the result asynchronously.
+        /// </summary>
+        /// <param name="cancellationToken">Cancellation token that can be used to cancel the operation.</param>
+        /// <returns>
+        /// Task containing the result of the operation: true if the reader was successfully advanced
+        /// to the next element; false if the reader has passed the end of the sequence.</returns>
+        Task<bool> MoveNext(CancellationToken cancellationToken);
```

Comment:
Is the goal to make this non-source breaking?

Reply:
Yeah

Reply:
my source got broken here, would it make sence to do something like 
`CancellationToken cancellationToken = CancellationToken.None` ?

Reply:
No, there's an extension method that provides the parameterless MoveNext():
https://github.com/grpc/grpc/pull/19059/files#diff-fa87174ba85fa18aefdd2f6f0c173750R39

you need to make sure that extension is visible from your code and you should be fine.

Reply:
@jtattermusch thanks for your answer, i'm using v 2.23. 
have `using Grpc.Core;`,
changed to `while (await responseStream.MoveNext(CancellationToken.None))` to make it work